source/common/upstream/load_balancer_impl.h

```diff
@@ -598,6 +600,29 @@ class LeastRequestLoadBalancer : public EdfLoadBalancerBase {
     initialize();
   }
 
+  LeastRequestLoadBalancer(
+      const PrioritySet& priority_set, const PrioritySet* local_priority_set, ClusterStats& stats,
+      Runtime::Loader& runtime, Random::RandomGenerator& random,
+      const envoy::config::cluster::v3::Cluster::CommonLbConfig& common_config,
+      const envoy::extensions::load_balancing_policies::least_request::v3::LeastRequest&
+          least_request_config,
+      TimeSource& time_source)
+      : EdfLoadBalancerBase(
+            priority_set, local_priority_set, stats, runtime, random, common_config,
+            (least_request_config.has_slow_start_config())
+                ? absl::optional<envoy::config::cluster::v3::Cluster::SlowStartConfig>(
+                      least_request_config.slow_start_config())
+                : absl::nullopt,
+            time_source),
+        choice_count_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(least_request_config, choice_count, 2)),
+        active_request_bias_runtime_(
+            least_request_config.has_active_request_bias()
+                ? absl::optional<Runtime::Double>(
+                      {least_request_config.active_request_bias(), runtime})
+                : absl::nullopt) {
+    initialize();
+  }
```

Comment:
Sorry for there is no specified related doc for it for now. I will clarify it it this PR or the original issue.

These are some short replies to your questions.

> The old config is still supported, right?

Yes, forever, unless one day we change our minds and decide to cut out v4 API.

> So shouldn't the old config create a synthetic typed config to allow for migration and then should that config be marked deprecated?

It will but not for now. This PR just make the `load_balancing_policy` work for least request policy. But It couldn't be used to replace original LB enum `LEAST_REQUEST` for now. There still some common problems need to be solved, e.g. the recreation of LB instance, subset support.
So I don't think we can mark the old configuration as deprecated yet. And as you said, it's will cause a huge amount of WARNING for end users, So I am not sure could we dreprecate it finally.

> I guess i'm trying to understand how old and new are going to coexist? Is there some detail somewhere that I can read? Thank you.

Both old config and new config will worked and the new config will be taken as a priority.